src/Diagram.hs

```diff
@@ -6,40 +6,43 @@ module Diagram where
 import Algebra.Graph.Labelled as LG
 import Control.Lens.TH (makeLenses)
 import FlowChart
+import GanttChart
 import Parser
-import Text.Megaparsec hiding (State)
 import Text.Megaparsec.Char (string)
 import qualified Text.Megaparsec.Char.Lexer as L
 
 data Diagram
   = FlowChart
       { _orientation :: Orientation,
-        _graph :: FlowChartGraph
+        _flowGraph :: FlowChartGraph
       }
-  | Others
+  | GanttChart {_ganttGraph :: GanttChartGraph}
```

Comment:
This is not bad:

```haskell
data = Diagram
  = D_FlowChart FlowChartGraph Orientation
  | D_GanttChart GanttChartGraph
```

(I'm curious why `Orientation` is not part of `FlowChartGraph`).

Reply:
The `FlowChartGraph`  is a type alias for `Graph (Maybe Edge) Node` , should I rename them to avoid confusion?

Reply:
You should just create a new record, and put the graph and orientation in it. Don't be shy of introducing new types (newtypes, data types). Type aliases should be used sparingly.

Reply:
Looks like you just did:

```haskell
data FlowChartGraph = FlowChartGraph
  { _flowChartOrientation :: Orientation,
    _flowChartGraph :: SubFlowChartGraph
  }
  deriving (Eq, Show, Generic)
```

But what is this "sub" flow chart? There is only one flow chart, no? Why not `_flowChartGraph :: Graph (Maybe Edge) Node`?

Reply:
Maybe this whole naming convention could be avoided if Diagram was defined like this:

```haskell
data Diagram = D_FlowChart FlowChart | D_GanttChart GanttChart
```

(ie., we are no longer misusing 'graph' terminology everywhere; and use it only where it is actually a graph)